lib/auth-providers/openid.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Refreshes a OpenID token.
+ */
+
+const Issuer = require('openid-client').Issuer;
+
+'use strict';
+
+module.exports = {
+  refresh: function (config) {
+    return new Promise((resolv, reject) => {
+      Issuer.discover(config['idp-issuer-url']) // => Promise
```

Comment:
do we need that `Promise` comment?

Reply:
Yeah it needs to be refactored, wrote this very quickly and it's going to be transformed into on-demand refresh in `request.js`. Thanks for the review!